docs/content/languages/nodejs.md

```diff
@@ -0,0 +1,420 @@
+```
+
+If everything was successful so far, then, build status of the source package will be set to 'succeeded'. This can be checked with the following command.
+
+```
+fission package info --name node-source-example-abcd
+```
+
+Next, test your function with the following and the output should have the current time.
+
+```
+fission fn test --name node-builder-example
+```
+
+#### Details of the fission nodejs builder image
+
+The builder has a build.sh script that performs an `npm install` of the node modules listed in user provided package.json. The builder image runs this script and packages the result into an archive.   
+When the function is invoked, one of the pods running the runtime image is specialized. What this means is that the archive created by the builder is fetched and extracted in the file system.
+Next, the user function is loaded according to the entry point specified with `fission fn create command`
+
+### Creating a custom nodejs builder image 
+
+If you'd like to do more than just `npm install` in the build step, you could customize the build.sh.
+Here's the link to the source code of fission [nodejs builder](https://github.com/fission/fission/tree/master/environments/nodejs/builder)
+
+As you can see, the build.sh performs a `npm install` inside a directory defined by the environment variable SRC_PKG and copies the built archive into a directory defined by environment variable DEPLOY_PKG 
+You could create a customized version of this build.sh with whatever additional commands needed to be run during the build step.
+
+Finally the image can be built with `docker build -t <USER>/nodejs-custom-builder .` and pushed to docker hub with `docker push <USER>/nodejs-custom-builder`
+
+Now you are ready to create a nodejs env with your custom builder image supplied to `--builder` flag
+
+## Modifying the nodejs runtime image
+
+If you wish to modify the nodejs runtime image to add more dependencies without using/creating a builder image, you can do so too.
+
+Here's the link to the source code of fission [nodejs runtime](https://github.com/fission/fission/tree/master/environments/nodejs)
+
+As you can see, there is a package.json in the directory with a list of node modules listed under dependencies section. 
+You can add the node modules required to this list and then build the docker image with `docker build -t <USER>/nodejs-custom-runtime .` and push the image `docker push <USER>/nodejs-custom-runtime`
+
+You are now ready to create a nodejs env with your image supplied to `--image` flag
+
+## Resource usage 
+
+Currently the nodejs environment containers are run with default memory limit of 512 MiB and a memory request of 256 MiB. Also, a default CPU limit of 1 and a CPU request of 0.5 cores.
+
+If you wish to create functions with higher resource requirements, you could supply `--mincpu`, `--maxcpu`, `--minmemory` and `--maxmemory` flags during `fission fn create`. Also supply `--executortype newdeploy` to the CLI.
```

Comment:
Wait why supply newdeploy?  They may want to use the pool?  You could just link to the executor doc here.